### DISCUSSION TITLE
Forums: Delete Post

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAWeb#654] [Ideas]
Forums: 
Add the ability for admin to Delete specific forum posts
For users to delete their own post.

**Comment by meleu** [comment]
IMO forum posts shouldn't be deleted. They should be editable by admins/moderators or the author. And a forum topic closed by admins/moderators. But never deleted.

**Comment by luchaos** [comment]
i'd actually prefer admins/mods not being able to edit other users' content, except for the motto on the user page. users should edit their contents themselves.
we will have soft-deletes, so a user's deleted posts can be recovered and remain visible to the author/mods/admins. allowing it to be edited by others allows for impersonation that may not be visible to non-admins.
edit: not admins, "forum mods" rather. we will have more specialized roles for the individual areas eventually.

**Comment by meleu** [comment]
> i'd actually prefer admins/mods not being able to edit other users' content

how are we going to deal with posts with links to download illegal/copyrighted material on our forums, for example?

**Comment by luchaos** [comment]
the post would be (soft) deleted, so it's only visible to forum mods/the author.
the user should be reminded to not post such links in a takedown notice that is delivered to the user's inbox.
when the user corrected that, the post can be recovered from trash.

for one, that makes the author aware of what is allowed instead of someone else fixing it for them and it's documented who posted what.

if we still want forum mods to being able to edit other users' posts, there should be a mandatory note that will also be displayed on the post where the mod describes what was changed.
still, i don't think that's a good approach. either a post follows the rules or not.

keeping track of how often a user did not follow the terms also allows to make a decision on whether a user should be suspended or banned from the forums.

**Comment by meleu** [comment]
Alright. I never saw a forum with this approach but it sounds like a good one! To be honest, at a first glance it seems to be the right thing to do on our community. :wink:
